### PR TITLE
upgrade azure client

### PR DESCRIPTION
**Pull request by madelynnblue**

### CODE REVIEW

**Comment by mjibson** [comment]
@bdarnell would you mind commenting on the first PR? I'm concerned it's too lenient and may accidentally allow real leaks now. But overall we've been routinely annoyed by keep alives.

File: pkg/util/leaktest/leaktest.go [hunk 1]
```diff
@@ -55,6 +55,9 @@ func interestingGoroutines() map[int64]string {
 		if stack == "" ||
 			strings.Contains(stack, "github.com/cockroachdb/cockroach/pkg/util/log.init") ||
 			strings.Contains(stack, "github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger") ||
+			// Ignore HTTP keep alives
+			strings.Contains(stack, "internal/poll.runtime_pollWait") ||
```
Comment (dt) [inline pkg/util/leaktest/leaktest.go]: hm, so I think `runtime_pollWait` could just be any waiting network conn -- is is possible to && this one with something that looks for `persistConn` further down to assert the it really is the http keep alive one that is sitting in the pollWait?

**Review by bdarnell — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28565)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/util/leaktest/leaktest.go, line 59 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28565#-LJrDCAiaRTT1D6ZF-QB-r1-59:-LJtnhaq4_PnJYbNPt2H:bid53cc) ([raw file](https://github.com/cockroachdb/cockroach/blob/d49cc72b0439249f51409e172cc5d0a45a345a81/pkg/util/leaktest/leaktest.go#L59)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary><blockquote>

I added another condition to the read loop.
</blockquote></details>

In the upstream leak test from net/http that this is based on, `writeLoop` is specifically called out as a disallowed goroutine: https://github.com/golang/go/blob/542ea5ad91367d2b40589942cc5757a4d5f43f97/src/net/http/main_test.go#L118

Can we solve this by adding a call to `net/http.(*Transport).CloseIdleConnections` in the affected tests? I don't think it's too critical to clean up after our http clients here, but since we've managed to do it so far I'd rather fix this if we can instead of whitelisting more goroutines.

**Comment by mjibson** [comment]
RFAL.

Did some investigation. Calling CloseIdleConnections doesn't close http2 connections, which our S3 tests use, so that doesn't work. I've changed the leaktest whitelist to be the same as in the stdlib http func, because I think that's the safest to use since it isn't system dependent and works for both http and http2 connections. This is a kinda big hammer, but I don't think it's worth spending more time to make this better. It feels like it would be a lot of effort for very tiny payoff, and there's more important things to work on.

**Comment by mjibson** [comment]
@dt @bdarnell can I get another review of this? Would like it to go in before Monday.

**Review by bdarnell — COMMENTED** [review]
Leaktest part LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28565#-:-LK3Qo9z0Kqn_ZdUHC6B:b-na71dx)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Review by dt — APPROVED** [review]

**Pull request merged**